substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -84,30 +85,30 @@ impl WasmInstance for Instance {
 			return (Err(format!("call into the runtime method '{name}': failed to write the input payload into guest memory: {error}").into()), None);
 		}
 
-		let mut state = ();
-		let mut call_args = polkavm::CallArgs::new(&mut state, method_index);
-		call_args.args_untyped(&[data_pointer, raw_data_length]);
-
-		match self.0.call(Default::default(), call_args) {
+		match self
+			.0
+			.call_typed(&mut (), export.program_counter(), (data_pointer, raw_data_length))
+		{
 			Ok(()) => {},
-			Err(polkavm::ExecutionError::Trap(trap)) => {
+			Err(polkavm::CallError::Trap) => {
 				return (
-					Err(format!("call into the runtime method '{name}' failed: {trap}").into()),
+					Err(format!("call into the runtime method '{name}' failed: trap").into()),
 					None,
 				);
 			},
-			Err(polkavm::ExecutionError::Error(error)) => {
+			Err(polkavm::CallError::Error(error)) => {
 				return (
 					Err(format!("call into the runtime method '{name}' failed: {error}").into()),
 					None,
 				);
 			},
-			Err(polkavm::ExecutionError::OutOfGas) => unreachable!("gas metering is never enabled"),
-		}
+			Err(polkavm::CallError::NotEnoughGas) => unreachable!("gas metering is never enabled"),
+			Err(polkavm::CallError::User(_)) => unreachable!(),
```

Comment:
`CallError::User` is not unreachable anymore due to your other change.

Reply:
Ya, refined the error handling a bit. Good catch.